upgrading-to-mockito-3.md

```diff
@@ -0,0 +1,142 @@
+# Migrating to Mockito 3
+
+Mockito 3 aims to provide a consistent, type-safe API that adheres to the new
+type rules of Dart 2 (both in terms of type inference, and runtime rules).
+
+In order to provide this API, Mockito's implementation had to be completely
+changed. In the old days of Dart 1 (and before the Dart Dev Compiler), we could
+pass any old object anywhere, and no one would complain. For example:
+
+```dart
+when(cat.eatFood(argThat(contains('mouse')), hungry: any))...
+```
+
+In this code, Mockito is not passing a String as the first argument to
+`eatFood()`, nor a bool as `hungry:`, which is what the method expects.
+Instead, Mockito passes an ArgumentMatcher, a completely different object from
+a String or bool! This is no longer allowed in Dart 2.  As an interim solution,
+Mockito 1.0.0 sports an awkward `typed` function, which wraps other Mockito
```

Comment:
Is this true, was it just for Mockito 2.x, right?

Reply:
Mockito 1 is when we introduced `typed`. Mockito 2 is when you un-exported `spy`.